domain/consensus/processes/dagtraversalmanager/dagtraversalmanager_test.go

```diff
@@ -3,11 +3,109 @@ package dagtraversalmanager_test
 import (
 	"github.com/kaspanet/kaspad/domain/consensus"
 	"github.com/kaspanet/kaspad/domain/consensus/model/externalapi"
+	"github.com/kaspanet/kaspad/domain/consensus/model/testapi"
 	"github.com/kaspanet/kaspad/domain/consensus/utils/testutils"
 	"github.com/kaspanet/kaspad/domain/dagconfig"
 	"testing"
 )
 
+func TestBlockAtDepth(t *testing.T) {
+	testutils.ForAllNets(t, true, func(t *testing.T, params *dagconfig.Params) {
+		factory := consensus.NewFactory()
+		tc, tearDown, err := factory.NewTestConsensus(params, false,
+			"TestBlockAtDepth")
+		if err != nil {
+			t.Fatalf("NewTestConsensus: %s", err)
+		}
+		defer tearDown(false)
+
+		// "checkExpectedBlockChainDag" - compare BlockAtDepth to a loop on the SelectedChainIterator on a single chain DAG.
+		checkExpectedBlockChainDag := func(highHash *externalapi.DomainHash, depth uint64) {
+			currentBlock := highHash
+			currentBlockData, err := tc.GHOSTDAGDataStore().Get(tc.DatabaseContext(), currentBlock)
+			if err != nil {
+				t.Fatalf("GHOSTDAGDataStore().Get: %+v", err)
+			}
+			for i := uint64(0); i <= depth; i++ {
+				if currentBlockData.SelectedParent() == nil {
+					break
+				}
+				currentBlock = currentBlockData.SelectedParent()
+				currentBlockData, err = tc.GHOSTDAGDataStore().Get(tc.DatabaseContext(), currentBlock)
+				if err != nil {
+					t.Fatalf("GHOSTDAGDataStore().Get: %+v", err)
+				}
+			}
+			expected := currentBlock
+			blockHash, err := tc.DAGTraversalManager().BlockAtDepth(highHash, depth)
+			if err != nil {
+				t.Fatalf("BlockAtDepth: %+v", err)
+			}
+
+			if !blockHash.Equal(expected) {
+				t.Fatalf("Expected block %s but got %s", expected, blockHash)
+			}
+		}
+
+		// "checkExpectedBlockTwoChildrenSameSelectedParent" checks blockAtDepth on 2 children with the same selectedParent.
+		checkExpectedBlockTwoChildrenSameSelectedParent := func(firstChild, secondChild *externalapi.DomainHash, depth uint64) {
+			blockHash, err := tc.DAGTraversalManager().BlockAtDepth(firstChild, depth)
+			if err != nil {
+				t.Fatalf("BlockAtDepth: %+v", err)
+			}
+			expectedSameHash, err := tc.DAGTraversalManager().BlockAtDepth(secondChild, depth)
+			if err != nil {
+				t.Fatalf("BlockAtDepth: %+v", err)
+			}
+			if !blockHash.Equal(expectedSameHash) {
+				t.Fatalf("Expected block %s but got %s", expectedSameHash, blockHash)
+			}
+		}
+
+		// "checkExpectedBlockTwoChildrenSameSelectedParent" checks on different chains, on the same DAG, that they merge at the correct point.
```

Comment:
This comment is broken I think... can't understand what it means and the name of the function is wrong

Reply:
Also, We try as hard as possible to never use abbreviations, because those are very ambiguous.

Specifically here, "Diff" is only used for "UTXODiffs", as in Difference. Using Diff for "Different" is confusing - so better just use the full word

Reply:
Changed, thanks.